internal/controlplane/server_test.go

```
@@ -354,6 +356,110 @@ func TestHeartbeatEndpointAuthenticatesAndRejectsInvalidLeases(t *testing.T) {
+		}
+		definitionPath := filepath.Join(directory, "config.toml")
+		if err := os.WriteFile(definitionPath, []byte("[agents.plan]\nexecutor = \"test\"\nprompt_file = \"plan.md\"\ntimeout = \"1m\"\n"), 0o600); err != nil {
+			t.Fatal(err)
+		}
+		store := openTestStore(t, filepath.Join(directory, "machinist.db"))
+		server, err := NewServer(store, definitionPath, "secret")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		probe, err := net.Listen("tcp", "127.0.0.1:0")
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)